elements/campaign-display/components/dfp-pixel.test.js

```diff
@@ -1,61 +1,94 @@
-import { createRenderer } from 'react-addons-test-utils';
 import React from 'react';
+import ReactDOM from 'react-dom';
 
 import DfpPixel from './dfp-pixel';
 
 describe('<campaign-display> <DfpPixel>', () => {
 
-  let shallowRenderer = createRenderer();
+  let reactContainer;
 
-  context('ad unit name', () => {
-
-    it('should always be "campaign-pixel"', function () {
-
-      shallowRenderer.render(<DfpPixel placement='junk' campaignId={1} />);
-
-      let html = shallowRenderer.getRenderOutput();
-      expect(html.props['data-ad-unit']).to.equal('campaign-pixel');
-    });
+  beforeEach(() => {
+    reactContainer = document.createElement('react-container');
+    document.body.appendChild(reactContainer);
   });
 
-  context('targeting parameters', () => {
-
-    it('should include ad unit placement', function () {
-      let placement = 'top';
-
-      shallowRenderer.render(<DfpPixel campaignId={1} placement={ placement } />);
-
-      let html = shallowRenderer.getRenderOutput();
-      expect(JSON.parse(html.props['data-targeting']).dfp_placement).to.equal(placement);
-    });
-
-    it('should require ad unit placement', function () {
-      chai.spy.on(console, 'error');
-
-      shallowRenderer.render(<DfpPixel campaignId={1} />);
+  afterEach(() => {
+    reactContainer.remove();
+  });
 
-      expect(console.error).to.have.been.called.with(
-        'Warning: Failed propType: Required prop `placement` was not specified in `DfpPixel`.'
+  context('on render', () => {
+
+    it('should call a callback', () => {
+      window.my = {
+        function: {
+          callback: chai.spy(),
+        },
+      };
+
+      let subject = ReactDOM.render(
+        <DfpPixel
+            placement='junk'
+            campaignId={1}
+            onRender='my.function.callback' />,
+        reactContainer
       );
-    });
-
-    it('should include campaign id', function () {
-      let id = 1;
 
-      shallowRenderer.render(<DfpPixel campaignId={ id } placement='junk' />);
-
-      let html = shallowRenderer.getRenderOutput();
-      expect(JSON.parse(html.props['data-targeting']).dfp_campaign_id).to.equal(id);
+      expect(window.my.function.callback)
+        .to.have.been.called.with(subject.refs.container);
     });
+  });
 
-    it('should require campaign id', function () {
-      chai.spy.on(console, 'error');
-
-      shallowRenderer.render(<DfpPixel placement='junk' />);
 
-      expect(console.error).to.have.been.called.with(
-        'Warning: Failed propType: Required prop `campaignId` was not specified in `DfpPixel`.'
-      );
-    });
-  });
+// TODO : add these back
```

Comment:
Just a POC at the moment, I'll add these back in when I merge back into https://github.com/theonion/bulbs-elements/pull/32.